File: ts_scripts/ai_competition_scraper.ts
import axios from 'axios';
import * as cheerio from 'cheerio';
import * as fs from 'fs';

async function scrapeAICompetition(): Promise<{ [key: string]: string }> {
    const companies = ["OpenAI", "xAI", "Gemini", "Anthropic"];
    const results: { [key: string]: string } = {};

    for (const company of companies) {
        const url = `https://www.google.com/search?q=${company}+latest+LLM+version`;
        try {
            const response = await axios.get(url);
            const $ = cheerio.load(response.data);
            
            // Extract the latest LLM version from the search results
            const latestVersion = $('.BNeawe').first().text() || 'Not found';
            results[company] = latestVersion;
        } catch (error) {
            console.error(`Error fetching data for ${company}:`, error);
            results[company] = 'Error';
        }
    }

    return results;
}

async function main() {
    const data = await scrapeAICompetition();
    const report = `# AI Competition Report\n\n` +
        Object.entries(data).map(([company, version]) => 
            `## ${company}\nLatest LLM Version: ${version}\n\n`
        ).join('');

    fs.writeFileSync('report.md', report, 'utf-8');
}

main().catch(console.error);

File: doc_crawler_minimal/ai_competition_scraper.py
import requests
from bs4 import BeautifulSoup
import json

def scrape_ai_competition():
    companies = ["OpenAI", "xAI", "Gemini", "Anthropic"]
    results = {}

    for company in companies:
        url = f"https://www.google.com/search?q={company}+latest+LLM+version"
        response = requests.get(url)
        soup = BeautifulSoup(response.text, "html.parser")
        
        # Extract the latest LLM version from the search results
        latest_version = soup.find("div", class_="BNeawe").text if soup.find("div", class_="BNeawe") else "Not found"
        results[company] = latest_version

    return results

if __name__ == "__main__":
    data = scrape_ai_competition()
    with open("report.md", "w", encoding="utf-8") as f:
        f.write("# AI Competition Report\n\n")
        for company, version in data.items():
            f.write(f"## {company}\n")
            f.write(f"Latest LLM Version: {version}\n\n")
